src/main/java/io/stargate/sgv2/jsonapi/api/request/EmbeddingProviderResponseValidation.java

```diff
@@ -0,0 +1,51 @@
+package io.stargate.sgv2.jsonapi.api.request;
+
+import static io.stargate.sgv2.jsonapi.exception.ErrorCode.EMBEDDING_PROVIDER_INVALID_RESPONSE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.client.ClientRequestContext;
+import jakarta.ws.rs.client.ClientResponseContext;
+import jakarta.ws.rs.client.ClientResponseFilter;
+import jakarta.ws.rs.core.MediaType;
+
+/**
+ * A client response filter/interceptor that validates the response from the embedding provider.
+ *
+ * <p>This filter checks the Content-Type of the response to ensure it is compatible with
+ * 'application/json' or 'text/json'. It also verifies the presence of a JSON body in the response.
+ *
+ * <p>If the response fails the validation, a {@link JsonApiException} is thrown with an appropriate
+ * error message.
+ */
+public class EmbeddingProviderResponseValidation implements ClientResponseFilter {
+  @Inject ObjectMapper objectMapper;
```

Comment:
Is this needed at this point? (I don't see usage but maybe I am missing something)

Reply:
I think so, 
basically it is the first gate of check when we get response from embedding providers.

Reply:
Oh, sorry, I misunderstand your point.
Yes, objectMapper could be deleted.

Reply:
Oh I discussed with Mahesh about the additional json deserialization and he thought it was unnecessary. Forgot to clear the Injection https://github.com/stargate/data-api/pull/979#discussion_r1535627134